corehq/blobs/fsdb.py

```diff
@@ -75,20 +88,29 @@ def delete(self, name=None, bucket=DEFAULT_BUCKET):
             path = safejoin(self.rootdir, bucket)
             remove = shutil.rmtree
         else:
-            path = self._getpath(name, bucket)
+            path = self.get_path(name, bucket)
             remove = os.remove
         if not exists(path):
             return False
         remove(path)
         return True
 
-    def _getpath(self, name, bucket):
+    @staticmethod
+    def get_unique_name(basename):
+        if SAFENAME.match(basename) and "/" not in basename:
+            prefix = basename
+        else:
+            prefix = "unsafe"
+        return prefix + "." + uuid4().hex
```

Comment:
Just thinking about how we could re-use this for the SQL form and case models and it would be easier if the base name could be empty and the returned ID would just be the UUID (wich is what we store in the DB as the identifier)